package-system/libpng/build_config.json

```diff
@@ -0,0 +1,49 @@
+{
+    "git_url": "https://git.code.sf.net/p/libpng/code",
+    "git_tag": "v1.6.37",
+    "package_name": "libpng",
+    "package_version": "1.6.37",
+    "package_url": "http://www.libpng.org/",
+    "package_license": "libpng-2.0",
+    "package_license_file": "LICENSE",
+    "cmake_find_source": "Findlibpng.cmake",
+    "cmake_find_target": "Findlibpng.cmake",
+    "Platforms": {
+        "Windows": {
+            "Windows": {
+                "cmake_generate_args": [
+                    "-DZLIB_ROOT=../../zlib-windows/zlib",
```

Comment:
Does this require you to have manually build zlib first?

Reply:
@nemerle  did bring up the possibility of using the [std_image_write.h](https://github.com/nothings/stb/blob/master/stb_image_write.h) header for writing pngs in this [comment](https://github.com/o3de/o3de/pull/4006#issuecomment-915940117)

That would avoid the explicit dependency on zlib.

Reply:
depending on zlib like this realyl isn't a big deal though.  We build it and the new one as of last week is cmake findzlib compatible.   The only issue is having to build it first.

Reply:
My suggestion was mostly related to resulting object-file sizes :smile:

Reply:
The only side-effect is that if you forget to build zlib locally some platforms will use the system library zlib copy. Mac & iOS both work without building zlib.

Reply:
I see.  We've actually had issues with that before.   For example, on mac, XCode ships with a LZMA library, so libtiff would detect and use it automatically. But, because it doesn't actually specify to _link_ to it, you end up with a linker error about missing lzma symbols.  I personally prefer to control the dependencies exactly (either use O3DE's real library, or don't use any library version at all, with the latter being the better choice if possible.)

Reply:
I stated in the readme.txt that you have to build the o3de zlib package locally first. This is how I built the Windows package. Is there a way we can prevent libpng from using some other zlib it finds? (I can investigate this if you all don't know off hand).

Reply:
it kind of depends.  I assume libpng uses cmake to build and not a custom make, if so, cmake accepts this

(libname)_DIR=(folder to look) to override where it looks for 3p

so, eg, adding -DZLIB_DIR=../../zlib-windows/zlib would do it.  Note that we want the 'zlib' subfolder, not the root, so that it uses the FindZLIB.cmake file in there, not the o3de "Findzlib.cmake" in the root.

Reply:
Okay, since this is using CMP0074=NEW and -DZLIB_ROOT=../../zlib-windows/zlib it sounds like we have what we need.